mewarpx/mewarpx/diags_store/flux_diagnostic.py

```diff
@@ -669,6 +669,37 @@ def __init__(self, diag_steps, runinfo,
             **kwargs
         )
 
+        # if this run is from a restart, try to load flux data up to the
+        # current point
+        if mwxrun.restart:
+            # the current simulation step presumably matches the restart step
```

Comment:
I would stick this whole section into a class function that's called here, for code readability.